lib/vmpooler/api/v1.rb

```diff
@@ -375,8 +390,16 @@ def atomically_allocate_vms(payload)
       content_type :json
       result = { 'ok' => false }
 
-      if jdata and !jdata.empty?
-        result = atomically_allocate_vms(jdata)
+      if jdata
+        empty = jdata.delete('empty')
+        invalid = jdata.delete('invalid')
+        statsd.increment(statsd_prefix + '.checkout.empty', empty) if !empty.nil?
+        statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if !invalid.nil?
+        if !jdata.empty?
+          result = atomically_allocate_vms(jdata)
+        else
```

Comment:
If you lose this `else; status 404`, does the next `else` (on line 404) trigger and return 404 instead?